ace_testbed/defiant_quick_start_guide.rst

```diff
@@ -0,0 +1,1142 @@
+
+MPI
+---
+
+The MPI implementation available on Defiant is Cray's MPICH, which is "GPU-aware"
+so GPU buffers can be passed directly to MPI calls. Currently, Defiant has MPICH
+versions 8.1.27 as default and 8.1.26 as an additional module.
+
+----
+
+Compiling
+=========
+
+This section covers how to compile for different programming models using the
+different compilers covered in the previous section.
+
+MPI
+---
+
++----------------+----------------+-----------------------------------------------------+-------------------------------------------------------------------------------+
+| Implementation | Module         | Compiler                                            | Header Files & Linking                                                        | 
++================+================+=====================================================+===============================================================================+
+| Cray MPICH     | ``cray-mpich`` | ``cc``, ``CC``, ``ftn`` (Cray compiler wrappers)    | MPI header files and linking is built into the Cray compiler wrappers         |
+|                |                +-----------------------------------------------------+-------------------------------------------------------------------------------+
+|                |                | ``hipcc``                                           | | ``-L$(MPICH_DIR)/lib -lmpi``                                                |
+|                |                |                                                     | | ``-I$(MPICH_DIR)/include``                                                  |
++----------------+----------------+-----------------------------------------------------+-------------------------------------------------------------------------------+
+
+
+GPU-Aware MPI
+^^^^^^^^^^^^^
+
+To use GPU-aware Cray MPICH, there are currently some extra steps needed in
+addition to the table above, which depend on the compiler that is used.
+
+1. Compiling with the Cray compiler wrappers, ``cc`` or ``CC``
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+To use GPU-aware Cray MPICH with the Cray compiler wrappers, users must load
+specific modules, set some environment variables, and include appropriate headers
+and libraries. The following modules and environment variables must be set:
+
+.. note:: 
+
+    Setting ``MPICH_SMP_SINGLE_COPY_MODE=CMA`` is required as a temporary
+    workaround due to a `known issue <https://docs.olcf.ornl.gov/systems/defiant_quick_start_guide.html#olcfdev-138-gpu-aware-cray-mpich-can-cause-hang-in-some-codes>`__.
+    Users should make a note of where they set this environment variable
+    (if e.g., set in a script) since it should NOT be set once the known
+    issue has been resolved.
+
+.. code:: bash
+
+    module load craype-accel-amd-gfx908
+    module load PrgEnv-cray
+    module load amd
+
+    ## These must be set before running
+    export MPIR_CVAR_GPU_EAGER_DEVICE_MEM=0
```

Comment:
Are these env variables still necessary to use GPU aware MPI? I'm not 100% sure that GPU aware MPI even works correctly on Defiant yet when I tried, but I might be doing it wrong. That's something to verify.

Reply:
Leaving for now until confirmed.